test/checks/mobile/css-orientation-lock.js

```diff
@@ -0,0 +1,243 @@
+		var actual = checks['css-orientation-lock'].evaluate.call(
+			checkContext,
+			document,
+			{},
+			undefined,
+			{
+				cssom: [
+					{
+						shadowId: 'a',
+						sheet: sheet
+					}
+				]
+			}
+		);
+		assert.isTrue(actual);
+	});
+
+	it('returns true if there are no ORIENTATION rule(s) within MEDIA rules in CSSOM stylesheets', function() {
+		var actual = checks['css-orientation-lock'].evaluate.call(
+			checkContext,
+			document,
+			{},
+			undefined,
+			{
+				cssom: [
+					{
+						shadowId: undefined,
+						sheet: getSheet(SHEET_DATA.BODY_STYLE)
+					},
+					{
+						shadowId: 'a',
+						sheet: getSheet(SHEET_DATA.MEDIA_STYLE_NON_ORIENTATION)
+					}
+				]
+			}
+		);
+		assert.isTrue(actual);
+	});
+
+	it('returns true if no styles within any of the ORIENTATION rule(s)', function() {
+		var actual = checks['css-orientation-lock'].evaluate.call(
+			checkContext,
+			document,
+			{},
+			undefined,
+			{
+				cssom: [
+					{
+						shadowId: undefined,
+						sheet: getSheet(SHEET_DATA.BODY_STYLE)
+					},
+					{
+						shadowId: 'a',
+						sheet: getSheet(SHEET_DATA.MEDIA_STYLE_ORIENTATION_EMPTY)
+					}
+				]
+			}
+		);
+		assert.isTrue(actual);
+	});
+
+	it('returns true if there is no TRANSFORM style within any of the ORIENTATION rule(s)', function() {
+		var actual = checks['css-orientation-lock'].evaluate.call(
+			checkContext,
+			document,
+			{},
+			undefined,
+			{
+				cssom: [
+					{
+						shadowId: 'a',
+						sheet: getSheet(
+							SHEET_DATA.MEDIA_STYLE_ORIENTATION_WITHOUT_TRANSFORM
+						)
+					}
+				]
+			}
+		);
+		assert.isTrue(actual);
+	});
+
+	it('returns true if TRANSFORM style applied is not ROTATE', function() {
+		var actual = checks['css-orientation-lock'].evaluate.call(
+			checkContext,
+			document,
+			{},
+			undefined,
+			{
+				cssom: [
+					{
+						shadowId: undefined,
+						sheet: getSheet(
+							SHEET_DATA.MEDIA_STYLE_ORIENTATION_WITH_TRANSFORM_NOT_ROTATE
+						)
+					}
+				]
+			}
+		);
+		assert.isTrue(actual);
+	});
+
+	it('returns true if TRANSFORM style applied is ROTATE, but is divisible by 180', function() {
+		var actual = checks['css-orientation-lock'].evaluate.call(
+			checkContext,
+			document,
+			{},
+			undefined,
+			{
+				cssom: [
+					{
+						shadowId: 'a',
+						root: document,
+						sheet: getSheet(
+							SHEET_DATA.MEDIA_STYLE_ORIENTATION_WITH_TRANSFORM_ROTATE_180
+						)
+					}
+				]
+			}
+		);
+		assert.isTrue(actual);
+	});
+
+	it('returns false if TRANSFORM style applied is ROTATE, and is divisible by 90 and not divisible by 180', function() {
+		var actual = checks['css-orientation-lock'].evaluate.call(
+			checkContext,
+			document,
+			{},
+			undefined,
+			{
+				cssom: [
+					{
+						shadowId: undefined,
+						root: document,
+						sheet: getSheet(
+							SHEET_DATA.MEDIA_STYLE_ORIENTATION_WITH_TRANSFORM_ROTATE_90
+						)
+					}
+				]
+			}
+		);
+		assert.isFalse(actual);
+	});
+
```

Comment:
Should test the related nodes.

Reply:
Done in integration tests below.

Reply:
Unit tests are intended to test the logic of your code. Use integration tests to make sure all of the pieces work together. Tests are better isolated that way.

Reply:
The reason related nodes are tested in integration as against unit, is I could not  find an example of how to get relatedNodes from check results with in unit tests. Any pointers on that will be appreciated.